toolchain/check/handle_binding_pattern.cpp

```diff
@@ -115,7 +115,7 @@ auto HandleBindingPattern(Context& context, Parse::NodeId parse_node) -> bool {
     }
 
     case Parse::NodeKind::ImplicitParamListStart:
-    case Parse::NodeKind::ParamListStart:
+    case Parse::NodeKind::TuplePatternStart:
       // Parameters can have incomplete types in a function declaration, but not
       // in a function definition. We don't know which kind we have here.
```

Comment:
This comment doesn't make a lot of sense any more. But I think this is deeper than the comment: the overall strategy of looking at the immediate context to determine whether you need a complete type -- or more broadly, what kind of instruction you're even creating -- doesn't make a lot of sense once we have nested patterns.

Maybe add a TODO to re-examine this?
```suggestion
      // Parameters can have incomplete types in a function declaration, but not
      // in a function definition. We don't know which kind we have here.
      // TODO: A tuple pattern can appear in other places than function parameters.
```